src/integration-test/kotlin/integration/WithdrawalIntegrationTest.kt

```diff
@@ -0,0 +1,205 @@
+package integration
+
+import com.github.kittinunf.result.failure
+import com.github.kittinunf.result.flatMap
+import config.EthereumPasswords
+import config.TestConfig
+import config.loadConfigs
+import contract.BasicCoin
+import jp.co.soramitsu.iroha.Keypair
+import jp.co.soramitsu.iroha.ModelCrypto
+import jp.co.soramitsu.iroha.ModelTransactionBuilder
+import kotlinx.coroutines.experimental.async
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.web3j.protocol.core.DefaultBlockParameterName
+import sidechain.eth.util.DeployHelper
+import sidechain.iroha.IrohaInitialization
+import sidechain.iroha.consumer.IrohaConsumerImpl
+import sidechain.iroha.consumer.IrohaNetworkImpl
+import sidechain.iroha.util.ModelUtil
+import util.getRandomString
+import java.math.BigInteger
+
+/**
+ * Integration tests for withdrawal usecase.
+ */
+class WithdrawalIntegrationTest {
+
+    init {
+        IrohaInitialization.loadIrohaLibrary()
+            .failure {
+                println(it)
+                System.exit(1)
+            }
+    }
+
+    /** Configurations for tests */
+    private val testConfig = loadConfigs("test", TestConfig::class.java)
+
+    /** Iroha transaction creator */
+    val creator = testConfig.iroha.creator
+
+    /** Iroha network layer */
+    val irohaConsumer = IrohaConsumerImpl(testConfig.iroha)
+
+    /** Ethereum password configs */
+    private val passwordConfig = loadConfigs("test", EthereumPasswords::class.java, "/ethereum_password.properties")
+
+    /** Ethereum utils */
+    private val deployHelper = DeployHelper(testConfig.ethereum, passwordConfig)
+
+    /** Ethereum test address */
+    private val toAddress = testConfig.ropstenTestAccount
+
+    /** Registration service port */
+    private val registrationServicePort = 8083
+
+    /** Notary account in Iroha */
+    val notaryAccount = testConfig.notaryIrohaAccount
+
+    /**
+     * Transfer asset in iroha with custom keypair
+     * @param creator - iroha transaction creator
+     * @param kp - keypair
+     * @param srcAccountId - source account id
+     * @param destAccountId - destination account id
+     * @param assetId - asset id
+     * @param description - transaction description
+     * @param amount - amount
+     * @return hex representation of transaction hash
+     */
+    fun transferAssetIroha(
```

Comment:
May be we should just call ```transferAssetIroha``` from ModelUtil?

Reply:
we can't, there is a custom keypair